runtime/lua/vim/treesitter/query.lua

```diff
@@ -370,8 +398,9 @@ local predicate_handlers = {
     return string_set[node_text]
   end,
 
+  -- |treesitter-predicate-has-ancestor?|
   ['has-ancestor?'] = function(match, _, _, predicate)
-    local node = match[predicate[2]]
+    local node = match[predicate[2]] ---@type TSNode?
```

Comment:
```suggestion
    local node = match[predicate[2] --[[@as integer]]]
```

Reply:
`[` `[[..]]` `]` is a bit ugly so I chose to use another local variable.